internal/signalio/type_test.go

```diff
@@ -102,3 +107,63 @@ func TestTypeUnmarshalText(t *testing.T) {
 		})
 	}
 }
+
+func TestWriterType_New(t *testing.T) {
```

Comment:
I'm not sold on changing the package from `signalio_test` to `signalio` to access the internal types to implement this test.

Some alternative approaches:
- check the result of `New(...)` against `JSONWriter(...)`
- update the `signalio.Writer` interface to include a `Type() signalio.WriterType` method
- call `WriteSignalSet(...)` with a `bytes.Buffer` and compare the output

Reply:
After taking a second look, I realized that replacing some struct calls would allow me to use the package `signalio_test` instead of `signalio`. Thanks!